app/controllers/events_controller.rb

```diff
@@ -3,6 +3,11 @@ class EventsController < ApplicationController
 
   def show
     @event = find_event(params[:id])
+    @donations = if params[:search]
```

Comment:
It acts somewhat as a facade, as it provides a single entry point into the application. 😀